File: frontend/src/api/guides.ts
import { apiRequest } from './client'

export type GuideStatus = 'GENERATING' | 'COMPLETED' | 'FAILED'

export type GuideReleaseDecision = 'PASS' | 'LIMITED' | 'REJECTED' | 'STALE'

export type GuideFallbackCode =
  | 'NO_APPROVED_EVIDENCE'
  | 'CONFLICTING_EVIDENCE'
  | 'PROVIDER_TIMEOUT'
  | 'DEPENDENCY_UNAVAILABLE'
  | 'VALIDATION_FAILED'
  | 'PRESCRIPTION_STALE'
  | 'EXECUTION_CONTEXT_STALE'
  | 'UNSUPPORTED_REQUEST'

export type GuideCitation = {
  source_type: 'LIFESTYLE_GUIDELINE'
  source_code: string
  source_version: string
  locator: string
  display_order: number
}

export type GuideData = {
  guide_id: string
  prescription_id: string
  prescription_version_id: string
  generation_status: GuideStatus
  content: string | null
  model_name: string | null
  prompt_version: string | null
  release_decision: GuideReleaseDecision | null
  release_is_current: boolean | null
  fallback_code: GuideFallbackCode | null
  fallback_text: string | null
  citations: GuideCitation[]
  requested_at: string
  completed_at: string | null
}

export type GuideResponse = {
  data: GuideData
}

export async function createGuide(
  prescriptionId: string,
): Promise<GuideResponse> {
  return apiRequest<GuideResponse>('/api/v1/guides', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      prescription_id: prescriptionId,
    }),
  })
}

export async function getGuide(guideId: string): Promise<GuideResponse> {
  return apiRequest<GuideResponse>(`/api/v1/guides/${guideId}`)
}

export async function getGuideForPrescription(
  prescriptionId: string,
): Promise<GuideResponse> {
  return apiRequest<GuideResponse>(
    `/api/v1/prescriptions/${prescriptionId}/guide`,
  )
}

File: backend/app/dtos/guides.py
from datetime import datetime
from enum import StrEnum
from uuid import UUID

from pydantic import BaseModel

from rag_runtime.guide_release_projection import (
    GuideRuntimeCitationSourceType,
    GuideRuntimeFallbackCode,
    GuideRuntimeReleaseDecision,
)


# GUIDE.generation_status와 동일한 값을 사용합니다.
class GuideStatus(StrEnum):
    GENERATING = "GENERATING"
    COMPLETED = "COMPLETED"
    FAILED = "FAILED"


class CreateGuideRequest(BaseModel):
    prescription_id: UUID


class GuideCitationData(BaseModel):
    source_type: GuideRuntimeCitationSourceType
    source_code: str
    source_version: str
    locator: str
    display_order: int


class GuideData(BaseModel):
    guide_id: UUID
    prescription_id: UUID
    prescription_version_id: UUID
    generation_status: GuideStatus
    content: str | None
    model_name: str | None
    prompt_version: str | None
    release_decision: GuideRuntimeReleaseDecision | None
    release_is_current: bool | None
    fallback_code: GuideRuntimeFallbackCode | None
    fallback_text: str | None
    citations: list[GuideCitationData]
    requested_at: datetime
    completed_at: datetime | None


class GuideResponse(BaseModel):
    data: GuideData
